### ISSUE TITLE
make sure that DESIRED_SITES requirement is an ordered list

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#7043]
With transition to python3 our PreJob.py insert a randomly-ordered list of sites in DESIRED_SITES in each job,
leading to explosion in the number of autoclusters.

see SubmissionInfrastructure MatterMost channel today https://mattermost.web.cern.ch/cms-o-and-c/pl/8z5zgt5m4t8wmj5h3tsyd5wxay
and in particular these posts [1] and private chat with Saqib [2]

[1]
https://mattermost.web.cern.ch/cms-o-and-c/pl/jrgmtuzdstde787q41fryie5de
https://mattermost.web.cern.ch/cms-o-and-c/pl/6ijokyyy538aby8ndo1na7s57o
https://mattermost.web.cern.ch/cms-o-and-c/pl/4hkgc5ubufr6pckr61mpta35no

[2]
https://mattermost.web.cern.ch/cms-o-and-c/pl/nsimt5xh4bnfpbzq41h4k9spec

**Comment by belforte** [comment]
code to fix is in
https://github.com/dmwm/CRABServer/blob/8e62371651b0ff8343ff07134424d97b48e8e954/src/python/TaskWorker/Actions/PreJob.py#L403-L440

and the problem is that when the site list is read from file and turned into a set in order to apply B/W masks, the order of the list is lost and the resulting list in https://github.com/dmwm/CRABServer/blob/8e62371651b0ff8343ff07134424d97b48e8e954/src/python/TaskWorker/Actions/PreJob.py#L438 can be in any random order. With python2, this order by chance was always the same, but with python3 every time PreJob.py is executed, a different list is generated.

consider this `test.py` example, where I stat from a list of 8 elements (like the 8 T2_US*)
```python
from  __future__ import print_function
t=['1','2','3','4','5','6','7','8']
av=set(t)
print(",".join(av))
```
and now let's run it multiple times only changing the interpreter:
```
belforte@vocms0194/tmp> for i in `seq 1000`; do python2 test.py; done|sort|uniq|wc -l
1
belforte@vocms0194/tmp> for i in `seq 1000`; do python3 test.py; done|sort|uniq|wc -l
991
belforte@vocms0194/tmp>
```

**Comment by belforte** [comment]
fixed in python3 branch via #7044 
tested with a 5-job task on preprod
